### DISCUSSION TITLE
Deployment Questions

### DISCUSSION THREAD

**Discussion by 0xHexE** [freshframework/fresh#406] [General]
So far, I like this project. It will be fantastic for ECommerce Platforms during the development of an e-commerce platform with the next platform. We found a lot of issues that fresh already resolve.

We are building one open-source ECommerce Platform. We want to try this framework. 

I saw the Deno deploy, which looks impressive, but I don't want to be any vendor lock when building a platform.

I have a couple of questions.

1. Can we Dockerize this application with the dependencies and lock file?  
2. Can we airgap the container after we build the image? We don't want to download dependencies after container is started. 
3. Can we build applications for older browsers. I saw that all scripts are modules. In some cases, we might need to support an old browser that doesn't support the module. Like legacy build system. If not, is this in the roadmap?
4. Any support for the CSP?

Thanks.

**Comment by vamshi9666** [comment]
Is it possible to deploy on elsewhere like a static cdn. 
I am hoping fresh consists of a command like "next export" in `next.js` to generate build at compile time. 
So far i didn't see any way to deploy anywhere but deno.deploy. 

I currently have next.js apps which are mostly static , which i am trying to migrate to something like astro.build for performance reasons. Since routing api of fresh is very similar to next.js, i thought migrating to fresh is an easy way. 
But unless i can build fresh app into static HTML `/dist` folder, i don't think it will help me.

**Comment by lucacasonato** [comment]
> Can we Dockerize this application with the dependencies and lock file?

Yes. You can `deno vendor` all dependencies.

> Can we airgap the container after we build the image? We don't want to download dependencies after container is started.

Yes, with the exception of the esbuild WASM (so island support). I am working on making fresh work completely air-gapped.

> Can we build applications for older browsers. I saw that all scripts are modules. In some cases, we might need to support an old browser that doesn't support the module. Like legacy build system. If not, is this in the roadmap?

This is not supported yet, but I am happy to merge PRs to support older browsers using a module/no-module pattern. #172 may be of interest to you.

> Any support for the CSP?

Yes. You can enable CSP in the route config, and then use the `useCSP` to add CSP exceptions. I still need to write docs for this though.

**Comment by rossholdway** [comment]
It'd be nice to have support for some of the older browsers. I think adding "Supported browsers" to the Fresh documentation would be a good start. Maybe like the Preactjs [supported browsers](https://preactjs.com/about/browser-support/) page?
